### PR TITLE
[SM-660] move sm-no-items to CL

### PR DESCRIPTION
**Pull request by willmartian**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

- moves the `sm-no-items` component to the CL
- renames to `bit-no-items`
- adds Storybook stories

## Code changes

- **bitwarden_license/bit-web/src/app/secrets-manager/*:** Renamed all instances of `sm-no-items`; updated module imports

## Screenshots

No visual changes.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/components/src/no-items/no-items.stories.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      imports: [ButtonModule, NoItemsModule],
+    }),
+  ],
+} as Meta;
+
+const Template: Story = (args) => ({
+  props: args,
+  template: `
+  <bit-no-items class="tw-text-main">
+    <ng-container slot="title">No items found</ng-container>
```
Comment (Hinton) [inline libs/components/src/no-items/no-items.stories.ts:20]: We should write a short section about slots on https://contributing.bitwarden.com/architecture/clients/presentation/angular.

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-accounts.module.ts [hunk 2]
```diff
@@ -19,7 +19,12 @@ import { ServiceAccountsRoutingModule } from "./service-accounts-routing.module"
 import { ServiceAccountsComponent } from "./service-accounts.component";
 
 @NgModule({
-  imports: [SecretsManagerSharedModule, ServiceAccountsRoutingModule, BreadcrumbsModule],
+  imports: [
+    SecretsManagerSharedModule,
+    ServiceAccountsRoutingModule,
+    BreadcrumbsModule,
+    NoItemsModule,
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-accounts.module.ts:26]: Seems we didn't have to import this for any of the other secret modules, is this needed?

File: libs/components/src/no-items/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export * from "./no-items.component";
```
Comment (Hinton) [inline libs/components/src/no-items/index.ts:1]: Do we need to export the component?

**Review by Hinton — COMMENTED** [review]
Looks good, a few small notes. And should we add some quick notes what the component is used for in the component file, which should show up in storybook?

```ts
/**
 * Component for displaying a message when there are no items to display. Expects a title,  description and button.
 */
```

File: libs/components/src/no-items/index.ts (hunk 3 above)
Comment (willmartian) [reply libs/components/src/no-items/index.ts]: Nope 👍

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-accounts.module.ts (hunk 2 above)
Comment (willmartian) [reply bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-accounts.module.ts]: The other components where it is used are declared in the shared SM module. I went ahead and just added `NoItemsModule` to the shared module's exports, so this is no longer needed.

**Comment by willmartian** [comment]
> Looks good, a few small notes. And should we add some quick notes what the component is used for in the component file, which should show up in storybook?
> 
> ```ts
> /**
>  * Component for displaying a message when there are no items to display. Expects title, description and button slots.
>  */
> ```

Tried adding this in a few different places and couldn't get it to show up in Storybook. It looks like that feature might come out [next release?](https://github.com/storybookjs/storybook/issues/8527#issuecomment-1298009627)

**Review by Hinton — DISMISSED** [review]
LGTM, thanks!

**Review by coltonhurst — APPROVED** [review]

**Pull request merged**